server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -60,7 +60,7 @@
 
     private static final ToXContent.Params FORMAT_PARAMS = new ToXContent.MapParams(Collections.singletonMap("pretty", "false"));
 
-    public static final int DEFAULT_PRE_FILTER_SHARD_SIZE = 128;
+    public static final int DEFAULT_PRE_FILTER_SHARD_SIZE = -1;
```

Comment:
nit: we serialize this value using `read/writeVInt` so `-1` is not a very good default since it will always use 5 bytes. This shouldn't affect anything but we also disallow setting this value to a negative number explicitly in `SearchRequest` so it will not be possible for users to restore the default value in a `SearchRequest`. I wonder if it would be simpler to use an `Integer` and `readOptionaVInt` to make the intent clear ?

Reply:
good point: I follow the bytes reasoning more than the resetting issue, cause generally we do not allow null values either when validating requests, hence you need to create a new request to go back to the default values. I am not a big fan of null values, so I have a slight preference for -1 but I can also change that